sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/CosmosClientConfiguration.scala

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.spark
+
+import java.lang.management.ManagementFactory
+
+private[spark] case class CosmosClientConfiguration (
+                                                      endpoint: String,
+                                                      key: String,
+                                                      applicationName: String,
+                                                      useGatewayMode: Boolean,
+                                                      useEventualConsistency: Boolean)
+
+private[spark] object CosmosClientConfiguration {
+  def apply(
+             config: Map[String, String],
+             useEventualConsistency: Boolean): CosmosClientConfiguration = {
+    val cosmosAccountConfig = CosmosAccountConfig.parseCosmosAccountConfig(config)
+    val applicationName = cosmosAccountConfig.applicationName match {
+      case None =>
+        s"${CosmosConstants.userAgentSuffix} ${ManagementFactory.getRuntimeMXBean.getName}"
```

Comment:
could you do a quick check if in all java jre this is safe to use this? is there any scenario that this can throw?

Reply:
I got the hint by looking at the current Spark connector https://github.com/Azure/azure-cosmosdb-spark/blob/1247559cae6b843a1ddc9fae9827222578abd42d/src/main/scala/com/microsoft/azure/cosmosdb/spark/ClientConfiguration.scala @FabianMeiswinkel what are your thoughts?

Reply:
It is safe in HDInsight and Databricks - I owld just surround it by try catch and have a fallback

Reply:
Makes sense, added try/catch